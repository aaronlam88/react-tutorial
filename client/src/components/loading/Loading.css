.center-page {
  position: absolute;
  top: 50%;
  left: 50%;
  z-index: 10;
}

.center-component {
  display: flex;
  align-items: center;
  justify-content: center
}