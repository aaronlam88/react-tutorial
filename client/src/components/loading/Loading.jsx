import React from 'react';

import CircularProgress from '@material-ui/core/CircularProgress';

import './Loading.css';

const LoadingPage = () => (<div className="center-page"><CircularProgress color="primary" /></div>);
const LoadingComponent = () => (<div className="center-component"><CircularProgress color="primary" /></div>);
const LoadingIcon = () => (<CircularProgress color="primary" />);

export { LoadingPage, LoadingComponent, LoadingIcon };
