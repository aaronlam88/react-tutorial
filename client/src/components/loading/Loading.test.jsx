import TestRender from "react-test-renderer";
import React from "react";
import { LoadingPage, LoadingComponent, LoadingIcon } from "./Loading";

it("should render without crashing", () => {
  const loadingPage = TestRender.create(<LoadingPage />).toJSON();
  expect(loadingPage).toMatchSnapshot();
  const loadingComponent = TestRender.create(<LoadingComponent />).toJSON();
  expect(loadingComponent).toMatchSnapshot();
  const loadingIcon = TestRender.create(<LoadingIcon />).toJSON();
  expect(loadingIcon).toMatchSnapshot();
});
