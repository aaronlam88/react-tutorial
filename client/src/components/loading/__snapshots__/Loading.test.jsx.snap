// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="center-page"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`should render without crashing 2`] = `
<div
  className="center-component"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`should render without crashing 3`] = `
<div
  className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
  role="progressbar"
  style={
    Object {
      "height": 40,
      "width": 40,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;
